src/component/plotly-graph/parse.js

```diff
@@ -108,4 +112,99 @@ function parseDim (result, opts, dim) {
   }
 }
 
+function willFigureHang (result) {
+  const data = result.figure.data
+
+  // cap the number of traces
+  if (data.length > 200) {
+    return true
+  }
+
+  const numberOfPtsPerTraceType = {}
+
+  for (let i = 0; i < data.length; i++) {
+    const trace = data[i] || {}
+    const type = trace.type || 'scatter'
+
+    if (!(type in numberOfPtsPerTraceType)) {
+      numberOfPtsPerTraceType[type] = 0
+    }
+
+    numberOfPtsPerTraceType[type] += estimateDataLength(trace)
+
+    // cap the number of point per trace type
+    if (numberOfPtsPerTraceType[type] > maxPtsPerTraceType(type)) {
+      return true
+    }
```

Comment:
I'm imagining someone *trying* to hang our servers, sending in one trace of each type, each just under our limit... do we want to do something like:
```js
budget += estimateDataLength(trace) / maxPtsPerTraceType(type);
if(budget > 1) return true;
```